apps/src/lib/node/ledger/shell/process_proposal.rs

```diff
@@ -42,11 +50,20 @@ where
     }
 
     /// Check all the given txs.
-    pub fn process_txs(&self, txs: &[Vec<u8>]) -> Vec<TxResult> {
-        let mut tx_queue_iter = self.storage.tx_queue.iter();
+    pub fn process_txs(&mut self, txs: &[Vec<u8>]) -> Vec<TxResult> {
```

Comment:
Unfortunately, we cannot make `self` mutable here because non-validator nodes are currently not required to run prepare/process proposal requests in TM and also we don't know if it's from an accepted proposal yet, so we should only mutate state once we're finalizing it

Reply:
I'll add a local cache in here and run the checks again in finalize block 👍